apps/storybook/src/stories/transfer-button/transfer-button.stories.tsx

```diff
@@ -0,0 +1,66 @@
+import { Label } from "@/components/ui/label";
+import { Meta, StoryObj } from "@storybook/react";
+import { useMemo, useState } from "react";
+import { Account } from "viem";
+import { withMockAccount, withWagmiProvider } from "../decorators/wagmi";
+import { TransferButton } from "./transfer-button";
+interface TransferButtonProps {
+	to: string;
+	account?: Account;
+}
+
+const TransferButtonStory = ({ to, account }: TransferButtonProps) => {
+	const [amount, setAmount] = useState("0.001");
+
+	const isValid = useMemo(() => {
+		return /([0-9]+\.)?[0-9]+$/.test(amount);
+	}, [amount]);
+
+	return (
+		<div className="flex flex-col gap-2">
+			<Label>Amount</Label>
+			<div className="flex items-center gap-2 relative border border-primary rounded-md pr-2 overflow-hidden">
+				<input
+					placeholder="0"
+					pattern="([0-9]+\.)?[0-9]+$"
+					value={String(amount)}
+					onKeyDown={(e) => {
+						if (e.key === "Backspace") {
+							setAmount(amount.slice(0, -1));
+						}
+					}}
+					onChange={(e) => {
+						const value = e.target.value;
+						// Only allow numbers and a single decimal point
+						if (/([0-9]+\.)?[0-9]+$/.test(value)) {
+							setAmount(value);
+						}
+					}}
+					className="border-none bg-white outline-none focus-within:outline-none px-4 focus:outline-none h-10 flex-1"
+				/>
```

Comment:
This does not handle cases where multiple characters are selected when pressing backspace.

Additionally, this is beginning to over-engineer the browser’s native behavior for this control, which may introduce unintended side effects. Now that isValid has been introduced, I think simply reverting to type="number" would resolve the UX issues.

```suggestion
        <input
          placeholder="0"
          type="number"
          value={String(amount)}
          onChange={(e) => setAmount(e.target.value)}
          className="border-none bg-white outline-none focus-within:outline-none px-4 focus:outline-none h-10 flex-1"
        />
```